src/Microsoft.Extensions.DependencyInjection/ServiceProvider.cs

```diff
@@ -188,6 +191,8 @@ public void Dispose()
 
         internal object CaptureDisposable(object service)
         {
+            _captureDisposableCallback?.Invoke(service);
```

Comment:
I didn't look at the ASM 😉 . Now we're down in the weeds. Branching vs virtual dispatch. It *does* clean up the test code since we don't need a derived service provider (that was kinda gross).